### PR TITLE
Munge Context Object Structure

### PR DESCRIPTION
**Pull request by lyzadanger**
This admittedly-mundane PR massages the shapes of the context objects being generated for patterns and pages so that I can move into beginning to output things. It's mostly concerned with creating correctly-nested and -keyed objects for later traversal and use. 

It's pretty boring.

But it was kind of hard.

zzzzzz recursion...

/cc @mrgerardorodriguez

### CODE REVIEW

File: src/options.js [hunk 1]
```diff
@@ -39,17 +39,19 @@ var parsers = {
...
-  parsers   : parsers,
-  partials  : 'src/partials/**/*',
-  patterns  : 'src/patterns/**/*'
+  layouts       : 'src/layouts/*',
+  markdownFields: ['notes'],
+  pages         : 'src/pages/**/*',
+  parsers       : parsers,
+  partials      : 'src/partials/**/*',
+  patterns      : 'src/patterns/**/*'
 };
```
Comment (lyzadanger) [inline src/options.js]: Sorry. This diff is just because alignment.

**Comment by lyzadanger** [comment]
Oh, yeah, it also introduces `markdownFields` to `options`. You can probably guess what that does.

File: src/parse.js [hunk 2]
```diff
@@ -1,7 +1,72 @@
...
+    delete fileObj.data;
+  }
+  return fileObj;
+}
+
+/**
+ * Build a single-page object for the page data object
+ */
+function pageEntry (pageFile, keys, options) {
+  const idKeys = keys.map(key => utils.keyname(key));
```
Comment (erikjung) [inline src/parse.js]: Nit: Could we simplify to this?

```js
const idKeys = keys.map(utils.keyname);
```

Comment (lyzadanger) [inline src/parse.js]: Good catch, sure.

**Comment by erikjung** [comment]
@lyzadanger One minor comment. :+1:

**Pull request merged**